api/util/parse_ibc_packet.go

```
@@ -0,0 +1,71 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/figment-networks/indexing-engine/structs"
+)
+
+var bigZero = new(big.Int).SetInt64(0)
+
+type PacketData struct {
+	Amount   string `json:"amount"`
+	Denom    string `json:"denom"`
+	Receiver string `json:"receiver"`
+	Sender   string `json:"sender"`
+}
+
+func ParsePacket(data []byte, event *structs.SubsetEvent) error {
+	var packetDataInterface interface{}
+	err := json.Unmarshal(data, &packetDataInterface)
+	if err != nil {
+		return fmt.Errorf("packet malformed: %w %s", err, string(data))
+	}
+	m, ok := packetDataInterface.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("packet malformed: %s", string(data))
+	}
+	if len(m) != 4 {
+		return fmt.Errorf("packet malformed: %s", string(data))
+	}
+	_, amtSok := m["amount"]
```

Are we sure that missing one of these keys means that a packet is malformed?  I think it might be better to accept what is given and only return an error in the case where Unmarshal returns an error.  We can log a warning if something doesn't look correct, but I don't think we want to return an error and discard the data or hold up the indexing because of it.